src/main/java/com/stuypulse/robot/commands/shooter/SetVelocityCommand.java

```diff
@@ -0,0 +1,24 @@
+package com.stuypulse.robot.commands.shooter;
+
+import com.stuypulse.robot.subsystems.Shooter;
+import com.stuypulse.stuylib.network.SmartNumber;
+
+import edu.wpi.first.wpilibj2.command.InstantCommand;
+
+public class SetVelocityCommand extends InstantCommand {
+
+    private final Shooter shooter;
+    private SmartNumber targetRPM;
+    
+    public SetVelocityCommand(Shooter shooter, SmartNumber targetRPM) {
```

Comment:
Probably replace SmartNumber with Number here. This will allow you to pass in double literals AND smartnumbers. You probably realized this but you do need smart numbers so every time you run this command it reads from the network, but Number seem to work fine in place (this appears in stuylib for hte same reason you have it in this command here)

If you have a Number you can give that to the shooter.targetRPM (this means replace setTargetRPM(SmartNumber) and setTargetRPM(double) with one method setTargetRPM(Number))

Reply:
/\ agreed

Reply:
does shooter.setShooterRPM even take in a smart number? otherwise this wont do what we think it will. Unless they do [SmartNumber.setNumber()](https://stuypulse.github.io/StuyLib/com/stuypulse/stuylib/network/SmartNumber.html#setNumber(java.lang.Number,java.lang.Number)) in the shooter subsystem. (dont it will cause problems) I think its a bit overkill to be passing smart numbers.